kernel/src/main/java/org/kframework/kompile/DefinitionParsing.java

```diff
@@ -277,63 +279,73 @@ protected Definition resolveConfigBubbles(Definition definition, Module defaultC
     }
 
     private Definition resolveConfigBubbles(Definition def) {
-      RuleGrammarGenerator gen = new RuleGrammarGenerator(def);
-      return DefinitionTransformer.from(m -> resolveConfigBubbles(def, m, gen), "parsing configs").apply(def);
-    }
-
-    private Module resolveConfigBubbles(Definition def, Module inputModule, RuleGrammarGenerator gen) {
-        if (stream(inputModule.localSentences())
-                .noneMatch(s -> s instanceof Bubble && ((Bubble) s).sentenceType().equals(configuration)))
-            return inputModule;
-
-        Set<Sentence> importedConfigurationSortsSubsortedToCell = stream(inputModule.productions())
-                .filter(p -> p.att().contains("cell"))
-                .map(p -> Production(Seq(), Sorts.Cell(), Seq(NonTerminal(p.sort())))).collect(Collections.toSet());
+        Definition defWithCaches = resolveCachedBubbles(def, false);
+        RuleGrammarGenerator gen = new RuleGrammarGenerator(def);
+
+        Definition defWithParsedConfigs = DefinitionTransformer.from(m -> {
+            if (stream(m.localSentences()).noneMatch(s -> s instanceof Bubble && ((Bubble) s).sentenceType().equals(configuration)))
+                return m;
+            Module ruleParserModule = gen.getConfigGrammar(m);
+
+            ParseCache cache = loadCache(ruleParserModule);
+            try (ParseInModule parser = RuleGrammarGenerator.getCombinedGrammar(cache.getModule(), isStrict, profileRules, files)) {
+                if (stream(m.localSentences()).filter(s -> s instanceof Bubble && ((Bubble) s).sentenceType().equals(configuration))
```

Comment:
We don't need this condition anymore, do we? We know there's at least one bubble that wasn't in the cache because we called `resolveCachedBubbles` and we still have configuration bubbles afterwards.